brave/src/main/java/com/linecorp/armeria/client/brave/ArmeriaHttpClientAdapter.java

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import java.net.SocketAddress;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestLogAvailability;
+import com.linecorp.armeria.internal.brave.SpanTags;
+
+import brave.http.HttpClientAdapter;
+
+public class ArmeriaHttpClientAdapter extends HttpClientAdapter<RequestLog, RequestLog> {
+    /**
+     * Original implementation is calling {@link HttpClientAdapter#url(Object)} which needs
+     * {@link RequestLog#scheme()} is not available at {@link RequestLogAvailability#REQUEST_START}.
+     * We need to use {@link RequestLog#path()} directly.
+     *
+     * @see brave.http.HttpAdapter#path(Object)
+     */
+    @Override
+    public String path(RequestLog requestLog) {
+        return requestLog.path();
+    }
+
+    @Override
+    public String method(RequestLog requestLog) {
+        return requestLog.method().name();
+    }
+
+    @Override
+    public String url(RequestLog requestLog) {
+        return SpanTags.generateUrl(requestLog);
+    }
+
+    @Override
+    @Nullable
+    public String requestHeader(RequestLog requestLog, String name) {
+        return requestLog.requestHeaders().get(name);
+    }
+
+    @Override
+    @Nullable
+    public Integer statusCode(RequestLog requestLog) {
+        final HttpStatus status = requestLog.status();
+        if (status == HttpStatus.UNKNOWN) {
+            return null;
+        }
+        return status.code();
+    }
+
+    /**
+     * Returns the authority of the {@link Request}.
+     */
+    public String authority(RequestLog requestLog) {
+        return requestLog.authority();
+    }
+
+    /**
+     * Returns the {@link SessionProtocol#uriText()} of the {@link Request}.
+     */
+    public String protocol(RequestLog requestLog) {
+        return requestLog.scheme().sessionProtocol().uriText();
+    }
+
+    /**
+     * Returns the {@link SerializationFormat#uriText()} if it's not {@link SerializationFormat#NONE}.
+     */
+    @Nullable
+    public String serializationFormat(RequestLog requestLog) {
+        final SerializationFormat serFmt = requestLog.scheme().serializationFormat();
+        return serFmt == SerializationFormat.NONE ? null : serFmt.uriText();
+    }
+
+    /**
+     * Returns the remote address.
+     */
+    @Nullable
+    public String remoteAddress(RequestLog requestLog) {
```

Comment:
do `HttpServerAdapter.parseClientIpAndPort` instead. not sure if this method is important enough to expose as in HttpClientParser someone can do this manually...